frame/assets/src/functions.rs

```diff
@@ -456,6 +456,7 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		}
 
 		let actual = Self::prep_debit(id, target, amount, f)?;
+		let mut target_died: DeadConsequence = Keep;
```

Comment:
I feel a bit weird using the type `DeadConsequence` here.
Because for me `DeadConsequence` is the the consequence of an account going below minimum balance of an asset.
(sidenote it also feel weird to me to call this type "dead" consequence considering that in case there is a held deposit then the account actually doesn't die at all AFAICT).

I feel it would be better to have it being a bool, WDYT ?

Reply:
I liked to use the `DeadConsequence` as return value of `transfer_and_die` since it has a `#[must_use]`.  
But in the cases like here it is probably better to use a `bool`, will change.  

I secretly hoped someone would find a better way than using a `mut bool`, it seems quite cumbersome.  
`Result.and_then` would not be nice as well, since it is nested in here and would also require helper vars.

Reply:
we can create a new type. `AccountStatus { Died, Alive }` maybe which is specifically design for the `FrozenBalance::died` callback.
As written in another comment I also find the current names confusing and not clearly defined. The term `Dead` is used in both `FrozenBalance::died` and `DeadConsequence` but they actually define a different concept of status.

Reply:
thinking again, maybe just use a `Option<DeadConsequence` is fine to me. then we assign some with the result of `dead_account`.
Because here `dead_account` is never called with `force` we can simply match `maybe_dead_consequence` to `Some(Remove)` and only execute the `FrozenBalance::died` in this case.

Reply:
Done